### DISCUSSION TITLE
Welcome to QuEST Discussions!

### DISCUSSION THREAD

**Discussion by TysonRayJones** [QuEST-Kit/QuEST#393] [Announcements]
## 👋 Welcome!
We’re testing using Discussions to keep track of what the community is working on, and what new features are coming to QuEST.

Feel free to create discussions which:
  * Describe new facilities you are working on which you hope to integrate into the QuEST repo.
  * Ask about new features, or propose new facilities.
 
This will, at least initially, be a very relaxed forum so feel free also to create discussions about anything you wish! You can of course also create an [Issue](https://github.com/QuEST-Kit/QuEST/issues).

**Comment by Elphikaus** [comment]
Hi, Great to see.

**Comment by ketan-more-github** [comment]
when i am using quest as backend GPU i am not able to saw tracess of usage of GPU using nvidia-smi cmd

**Reply by TysonRayJones** [reply]
Hi Ketan,

That likely indicates QuEST is _not_ running on the GPU. Have you compiled QuEST with GPU-acceleration enabled, as outlined [here](https://github.com/QuEST-Kit/QuEST/blob/master/examples/README.md#compiling)?

The upcoming [`v4`](https://github.com/QuEST-Kit/QuEST/tree/v4) (only days away! :tada:) conveniently provides a `reportQuESTEnv()` function to make ascertaining the utilised deployments much easier. You'd simply run:
```C++
reportQuESTEnv();
```
to see output like:
```C++
QuEST execution environment:
...
  [compilation]
    isMpiCompiled...........1
    isGpuCompiled...........1
    isOmpCompiled...........1
    isCuQuantumCompiled.....0
  [deployment]
    isMpiEnabled.....1
    isGpuEnabled.....1
    isOmpEnabled.....1
  [cpu]
...
  [gpu]
    numGpus...........4
    gpuDirect.........1
    gpuMemPools.......1
    gpuMemory.........79.3 GiB per gpu
    gpuMemoryFree.....78.8 GiB per gpu
    gpuCache..........0 bytes per gpu
  [distribution]
    isMpiGpuAware.....1
    numMpiNodes.......4
...
  [statevector autodeployment]
    8 qubits......[omp]
    12 qubits.....[omp] [gpu]
    28 qubits.....[omp] [gpu] [mpi]
  [density matrix autodeployment]
    4 qubits......[omp]
    6 qubits......[omp] [gpu]
    14 qubits.....[omp] [gpu] [mpi]
```
Or you can see the parallelisation automatically chosen for a particular `Qureg` with
```C++
reportQuregParams(qureg);
```
which might show
```C++
Qureg:
  [deployment]
    isMpiEnabled.....1
    isGpuEnabled.....1
    isOmpEnabled.....1
  [dimension]
    isDensMatr.....0
    numQubits......34
    numCols........N/A
    numAmps........2^34 = 17179869184
  [distribution]
    numNodes.....2^2 = 4
    numCols......N/A
    numAmps......2^32 = 4294967296 per node
  [memory]
    cpuAmps...........32 GiB per node
    gpuAmps...........32 GiB per node
    cpuCommBuffer.....32 GiB per node
    gpuCommBuffer.....32 GiB per node
    globalTotal.......512 GiB
```

If you're just getting started with a new project, I would recommend you trying the new [`v4`](https://github.com/QuEST-Kit/QuEST/tree/v4) branch. Setup is a little different - I'm happy to guide you through it while we polish the docs!

**Comment by ketan-more-github** [comment]
Yes i have compiled source with gpu and cu quantum also and i get effective
results. But didn't get insight on gpu card..for running process
No issue i will love to try V4 branch i will check it
